### ISSUE TITLE
unexpected EOF while parsing

### ISSUE THREAD

**Issue by BayuSasongko** [dingwanli/calculate-precision-recall-f1score-roc-auc#1]
I have problem with my code :
`img_width = 42                                                                      #parameter：the width of image        
img_height = 42                                                                      #parameter：the height of image
batch_size = 64

train_datagen = ImageDataGenerator(preprocessing_function=preprocess_input,
        width_shift_range=0.2, height_shift_range=0.2,
        rescale=1./255,
        zoom_range=0.25,
        horizontal_flip=True,vertical_flip=True,fill_mode="nearest")
test_datagen = ImageDataGenerator(preprocessing_function=preprocess_input,
        width_shift_range=0.2, height_shift_range=0.2,
        rescale=1./255, zoom_range=0.25,
        horizontal_flip=True,vertical_flip=True,fill_mode="nearest")
val_datagen = ImageDataGenerator(preprocessing_function=preprocess_input,
        width_shift_range=0.2, height_shift_range=0.2,
        rescale=1./255, zoom_range=0.25,
        horizontal_flip=True,vertical_flip=True,fill_mode="nearest")

train_gen=train_datagen.flow_from_directory(
        train_data_dir,
        target_size=(img_width,img_height),
        batch_size=batch_size,
        color_mode="rgb",
        class_mode='categorical')
test_gen=test_datagen.flow_from_directory(
        test_dir,
        target_size=(img_width,img_height),
        batch_size=batch_size,color_mode="rgb",
        class_mode='categorical')
val_gen=val_datagen.flow_from_directory(validation_data_dir,target_size=(img_width,img_height),
                                        batch_size=batch_size,color_mode="rgb",
                                        class_mode="categorical")

#generate the label of validate dataset.
num_withoutmask = len(os.listdir(withoutmask_dir))
withoutmask_label = np.zeros(num_withoutmask, dtype=np.int8)
print('withoutmask_label: ', withoutmask_label)
num_withmask = len(os.listdir(withmask_dir))
withmask_label = np.ones(num_withmask, dtype=np.int8)
print('withmask_label: ', withmask_label)
label = np.hstack((withoutmask_label, withmask_label))
print('label: ', label)

##generate the label of training dataset.
train_num_withoutmask = len(os.listdir(train_withoutmask_dir))
train_withoutmask_label = np.zeros(train_num_withoutmask, dtype=np.int8)
print('train_withoutmask_label:', train_withoutmask_label, 'number of train_withoutmask_label:', len(train_withoutmask_label))
train_num_withmask = len(os.listdir(train_withmask_dir))
train_withmask_label = np.ones(train_num_withmask, dtype=np.int8)
print('train_withmask_label:', train_withmask_label,  'number of train_withmask_label:', len(train_withmask_label))
train_label = np.hstack((train_withoutmask_label, train_withmask_label))
print('train_label:', train_label,  'number of train_label:', len(train_label))

withoutmask_predict = []
withmask_predict = []
train_withoutmask_predict = []
train_withmask_predict = []

train_loss_score = []
train_f1_score = []
train_recall_score = []
train_precision_score = []
train_acc = []
train_auc_score = []
val_loss = []
val_acc = []
val_f1_score = []
val_recall_score = []
val_precision_score = []
val_auc_score = []

class Evaluate(Callback):
    def __init__(self):
        super().__init__()
        
    def on_train_end(self, logs={}):
        withmask_predict = []
        withoutmask_predict = []
        train_withmask_predict = []
        train_withoutmask_predict = []
        
        #calculate the evaluation criteria of validate dateset
        for img_name in os.listdir(withmask_dir):
            img_path = os.path.join(withmask_dir, img_name)
            img = image.load_img(img_path, target_size=(img_width, img_height))
            x = image.img_to_array(img)
            x = np.expand_dims(x, axis=0)
            x = preprocess_input(x)
            withmask_result = self.model.predict(x)
            withmask_result = eval(str(withmask_result).strip('[]'))
            withmask_predict.append(withmask_result)
        withmask_predict_score = np.asanyarray(withmask_predict, dtype=float)
        withmask_predict1 = np.asanyarray(np.asarray(withmask_predict, dtype=float).round(),dtype=np.int8)     
        for img_name in os.listdir(withoutmask_dir):
            img_path = os.path.join(withoutmask_dir, img_name)
            img = image.load_img(img_path, target_size=(img_width, img_height))
            x = image.img_to_array(img)
            x = np.expand_dims(x, axis=0)
            x = preprocess_input(x)
            withoutmask_result = self.model.predict(x)
            withoutmask_result = eval(str(withoutmask_result).strip('[]'))
            withoutmask_predict.append(withoutmask_result)
        withoutmask_predict_score = np.asanyarray(withoutmask_predict, dtype=float)
        withoutmask_predict1 = np.asarray(np.asarray(withoutmask_predict, dtype=float).round(),dtype=np.int8)
        predict_score = np.hstack((withmask_predict_score, withoutmask_predict_score))
        predict = np.hstack((withmask_predict1, withoutmask_predict1))
#         print('predict_score', predict_score)
#         print('predict', predict)
        
        #calculate the evaluation criteria of training dateset
        for img_name in os.listdir(train_withmask_dir):
            img_path = os.path.join(train_withmask_dir, img_name)
            img = image.load_img(img_path, target_size=(img_width, img_height))
            x = image.img_to_array(img)
            x = np.expand_dims(x, axis=0)
            x = preprocess_input(x)
            train_withmask_result = self.model.predict(x)
            train_withmask_result = eval(str(train_withmask_result).strip('[]'))
            train_withmask_predict.append(train_withmask_result)
        train_withmask_predict_score = np.asanyarray(train_withmask_predict, dtype=float)
        train_withmask_predict1 = np.asanyarray(np.asarray(train_withmask_predict, dtype=float).round(),dtype=np.int8) 
        for img_name in os.listdir(train_withoutmask_dir):
            img_path = os.path.join(train_withoutmask_dir, img_name)
            img = image.load_img(img_path, target_size=(img_width, img_height))
            x = image.img_to_array(img)
            x = np.expand_dims(x, axis=0)
            x = preprocess_input(x)
            train_withoutmask_result = self.model.predict(x)
            train_withoutmask_result = eval(str(train_withoutmask_result).strip('[]'))
            train_withoutmask_predict.append(train_withoutmask_result)
        train_withoutmask_predict_score = np.asanyarray(train_withoutmask_predict, dtype=float)
        train_withoutmask_predict1 = np.asarray(np.asarray(train_withoutmask_predict, dtype=float).round(),dtype=np.int8)
        
        train_predict_score = np.hstack((train_withmask_predict_score, train_withoutmask_predict_score))
        train_predict = np.hstack((train_withmask_predict1, train_withoutmask_predict1))
#         print('train_predict_score', train_predict_score)
#         print('train_predict', train_predict, len(train_predict))
        
        tn, fp, fn, tp = confusion_matrix(label, predict).ravel()
        print('tn',tn)
        print('fp',fp)
        print('fn',fn)
        print('tp',tp)
        validate_loss = log_loss(label, predict)
        print('validate_loss', validate_loss)
        accuracy = accuracy_score(label, predict)
        print('accuracy', accuracy)
        precision = precision_score(label, predict,average='binary')
        print('precision', precision)      
        recall = recall_score(label, predict,average='binary')
        print('recall', recall)        
        f1 = f1_score(label, predict,average='binary')
        print('f1', f1)
        auc = roc_auc_score(label, predict_score)
        print('auc', auc)
        fpr, tpr, thresholds = roc_curve(label, predict_score, pos_label=1)
        
        train_tn, train_fp, train_fn, train_tp = confusion_matrix(train_label, train_predict).ravel()
        print('train_tn',train_tn)
        print('train_fp',train_fp)
        print('train_fn',train_fn)
        print('train_tp',train_tp)
        train_loss = log_loss(train_label, train_predict)
        print('train_loss', train_loss)
        train_accuracy = accuracy_score(train_label, train_predict)
        print('train_accuracy', train_accuracy)
        train_precision = precision_score(train_label, train_predict,average='binary')
        print('train_precision', train_precision)      
        train_recall = recall_score(train_label, train_predict,average='binary')
        print('train_recall', train_recall)        
        train_f1 = f1_score(train_label, train_predict,average='binary')
        print('train_f1', train_f1)
        train_auc = roc_auc_score(train_label, train_predict_score)
        print('train_auc', train_auc)
        train_fpr, train_tpr, train_thresholds = roc_curve(train_label, train_predict_score, pos_label=1)
        
        plt.figure(figsize=(5,5))     
        plt.plot(train_fpr,train_tpr,color='blue',linewidth=3,label='train_AUC (area:%0.4f)'%train_auc)  
        plt.plot(fpr,tpr,color='red',linewidth=3,label='validate_AUC (area:%0.4f)'%auc)  
        plt.plot([0, 1], [0, 1], color='black',linewidth=2, linestyle='--')  
        plt.xlim([0, 1])  
        plt.ylim([0, 1])  
        plt.xlabel('False Positive Rate')  
        plt.ylabel('True Positive Rate')  
        plt.title('ROC')  
        plt.legend(loc="lower right")      
        plt.show()
        
        train_loss_score.append(train_loss)
        train_acc.append(train_accuracy)
        train_precision_score.append(train_precision)
        train_recall_score.append(train_recall)
        train_f1_score.append(train_f1)
        train_auc_score.append(train_auc)
        val_loss.append(validate_loss)
        val_acc.append(accuracy)
        val_precision_score.append(precision)
        val_recall_score.append(recall)
        val_f1_score.append(f1)
        val_auc_score.append(auc)

MobileNet = MobileNet(weights='imagenet', include_top=False, input_shape=(42,42,3))

x = MobileNet.output
x = layers.GlobalAveragePooling2D()(x)
x = layers.Dense(42, activation='relu')(x)
prediction = Dense(2, activation='sigmoid')(x)
MobileNet_transfer = models.Model(inputs=MobileNet.input, outputs=prediction)

MobileNet_transfer.summary()`

is error : unexpected EOF while parsing

![image](https://user-images.githubusercontent.com/17606892/118157829-3af7e600-b445-11eb-84ef-f95c85e6d9cf.png)
